torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -405,6 +413,51 @@ std::string TensorPipeAgent::createUniqueShmAddr() {
 }
 #endif
 
+std::string TensorPipeAgent::getDefaultIPAddress() {
+  char hostname[HOST_NAME_MAX];
+  int rv = gethostname(hostname, HOST_NAME_MAX);
+  TORCH_INTERNAL_ASSERT(rv == 0, "Unable to get local hostname");
+
+  struct addrinfo hints, *servinfo, *p;
+  memset(&hints, 0, sizeof hints);
+  hints.ai_family = AF_UNSPEC;
+  hints.ai_socktype = SOCK_STREAM;
+  rv = getaddrinfo(hostname, nullptr, &hints, &servinfo);
+  TORCH_INTERNAL_ASSERT(rv == 0, "Get address info error: ", gai_strerror(rv));
```

Comment:
should we instead fallback if getaddrinfo() fails? Similar with gethostname()?

getaddrinfo() can be a bit flaky in prod, since it generates network traffic

Reply:
Does it generate DNS traffic even when resolving its own hostname?

Reply:
My guess is that it would go through DNS. Yeah we could fallback